src/Compilers/VisualBasic/Test/Semantic/Semantics/RequiredMembersTests.vb

```diff
@@ -0,0 +1,1842 @@
+                                        </expected>)
+        End Sub
+
+        <Fact>
+        Public Sub IndexedPropertyOverload_AllSet()
+            ' Equivalent to
+            ' <RequiredMember>
+            ' Public Class C
+            '     <RequiredMember>
+            '     Public Overloads Property P1 As Integer
+            '     Public Overloads Property P1(x As Integer) As Integer
+            '         Get
+            '             Return 0
+            '         End Get
+            '         Set
+            '         End Set
+            '     End Property
+            ' End Class
+            Dim il = "
+.class public auto ansi C
+    extends [mscorlib]System.Object
+{
+    .custom instance void [mscorlib]System.Runtime.CompilerServices.RequiredMemberAttribute::.ctor() = (
+        01 00 00 00
+    )
+    .field private int32 _P1
+
+    .method public specialname rtspecialname 
+        instance void .ctor () cil managed 
+    {
+        ldarg.0
+        call instance void [mscorlib]System.Object::.ctor()
+        ret
+    }
+
+    .method public hidebysig specialname 
+        instance int32 get_P1 (
+            int32 x
+        ) cil managed 
+    {
+        ldc.i4.0
+        ret
+    }
+
+    .method public hidebysig specialname 
+        instance void set_P1 (
+            int32 x,
+            int32 Value
+        ) cil managed 
+    {
+        ret
+    }
+
+    .method public hidebysig specialname 
+        instance int32 get_P1 () cil managed 
+    {
+        ldarg.0
+        ldfld int32 C::_P1
+        ret
+    }
+
+    .method public hidebysig specialname 
+        instance void set_P1 (
+            int32 AutoPropertyValue
+        ) cil managed 
+    {
+        ldarg.0
+        ldarg.1
+        stfld int32 C::_P1
+        ret
+    }
+
+    .property instance int32 P1(
+        int32 x
+    )
+    {
+        .get instance int32 C::get_P1(int32)
+        .set instance void C::set_P1(int32, int32)
+    }
+    .property instance int32 P1()
+    {
+        .custom instance void [mscorlib]System.Runtime.CompilerServices.RequiredMemberAttribute::.ctor() = (
+            01 00 00 00
+        )
+        .get instance int32 C::get_P1()
+        .set instance void C::set_P1(int32)
+    }
+}"
+
+            Dim ilRef = CompileIL(il)
+
+            Dim comp = CreateCompilation("
+Module M
+    Sub Main()
+        Dim c = New C() With { .P1 = 1 }
+    End Sub
+End Module", {ilRef}, targetFramework:=TargetFramework.Net70)
+
+            comp.AssertNoDiagnostics()
+        End Sub
+
+        <Fact>
+        Public Sub IndexedProperty_OverloadInDerivedType()
+            Dim csharpComp = CreateCSharpCompilationWithRequiredMembers("
+public class C1
+{
+    public required int P1 {get;set;}
+}
+
+public class C2 : C1
+{
+    [System.Runtime.CompilerServices.IndexerNameAttribute(nameof(P1))]
+    public int this[int x] => x;
+}")
+
+            Dim comp = CreateCompilation("
+Module M
+    Sub Main()
+        Dim c = New C2()
```

Comment:
>Dim c = New C2() [](http://example.com/codeflow?start=8&length=16)

I think it would be good to test success scenario as well, i.e. with initializer for P1.